src/main/java/Filters/LoginFilter.java

```diff
@@ -0,0 +1,48 @@
+package Filters;
+
+import javax.servlet.*;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+
+@WebFilter(filterName = "LoginFilter")
+public class LoginFilter implements Filter {
+    @Override
+    public void destroy() {}
+    @Override
+    public void init(FilterConfig config) throws ServletException {}
+
+    @Override
+    public void doFilter(ServletRequest req, ServletResponse resp, FilterChain chain) throws ServletException, IOException {
+        HttpServletRequest request = (HttpServletRequest) req;
+        HttpServletResponse response = (HttpServletResponse) resp;
+        HttpSession session = request.getSession(false);
+
+        boolean loggedIn = session != null && session.getAttribute("currentUser") != null;
+        boolean isLoginRequest = request.getServletPath().equals("/login");
+        boolean isRegistrationRequest = request.getServletPath().equals("/register");
+
+        // TODO: clean up this logic so its not cursed?
```

Comment:
Could you talk through this logic a bit too, please?

Reply:
Most websites will redirect a logged in user to their dashboard page, for example, Facebook will redirect you to your feed if you try to access its login page while you have a session. In order to replicate this, we check if the user is logged in, if they are, we check if they are trying to access /login or /register and redirect them to their dashboard if they are. If they are logged in and trying to access something else, we forward the request and give them access to whatever they are trying to get into as usual. If they are sending a POST request to /login, this is treated as them wishing to log OUT and thus allows them to perform the request as usual, resulting in them being logged out due to the code on the /login servlet.

Reply:
Here is a flowchart of my intended logic, hopefully this makes it clearer - the code is pretty nasty tbh
![image](https://user-images.githubusercontent.com/26003386/102026763-9ee38900-3d97-11eb-8094-d0261503c9f4.png)

Reply:
This does miss the non logged in user trying to access /register <.<

Reply:
Okay, flow chart 2 electric boogaloo:
![image](https://user-images.githubusercontent.com/26003386/102026957-fdf5cd80-3d98-11eb-9c93-174794a50c42.png)
Also in the making of this realised I missed an if check for the registration page while logged in, so thanks for asking!

Reply:
Similarly here, I suppose a named parameter in the POST would make this clearer? Instead of making those changes, maybe adding some comments in the code for this section would certainly help me understand.

Reply:
I'll think about adding a comment to the area when I'm done with the calendar stuff